### PR TITLE
MGMT-1141 - fix assisted-installer joined status reporting

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -262,21 +262,26 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
 		if err != nil {
 			i.log.Warnf("Still waiting for master nodes: %v", err)
-		} else {
-			i.updateReadyMasters(nodes, &readyMasters)
-			i.log.Infof("Found %d ready master nodes", len(readyMasters))
-			if len(readyMasters) >= minMasterNodes {
-				i.log.Infof("Waiting for master nodes - Done")
-				cancel()
-			}
+			return
```
Comment (tsorya) [inline src/installer/installer.go:278]: why return? it will not exit the func

File: src/installer/installer.go [hunk 2]
```diff
@@ -286,7 +291,19 @@ func (i *installer) waitForMasterNodes(ctx context.Context, minMasterNodes int,
 	return nil
 }
 
-func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters *[]string) {
+func (i *installer) getInventoryHostsMap(hostsMap map[string]string) map[string]string {
```
Comment (tsorya) [inline src/installer/installer.go:294]: don't we have same function in controller?

File: src/installer/installer_test.go [hunk 3]
```diff
@@ -49,6 +59,13 @@ var _ = Describe("installer HostRoleMaster role", func() {
 	downloadFileSuccess := func(fileName string) {
 		mockbmclient.EXPECT().DownloadFile(fileName, filepath.Join(InstallDir, fileName)).Return(nil).Times(1)
 	}
+	/*
```
Comment (tsorya) [inline src/installer/installer_test.go:62]: delete it?

File: src/installer/installer_test.go [hunk 4]
```diff
@@ -109,13 +126,17 @@ var _ = Describe("installer HostRoleMaster role", func() {
...
-			hostIds := []string{"7916fa89-ea7a-443e-a862-b3e930309f65"}
-			mockk8sclient.EXPECT().ListMasterNodes().Return(GetKubeNodes(hostIds), nil).Times(1)
-			mockbmclient.EXPECT().UpdateHostStatus(Joined, hostIds[0]).Times(1)
-			hostIds = []string{"7916fa89-ea7a-443e-a862-b3e930309f65", "eb82821f-bf21-4614-9a3b-ecb07929f238"}
-			mockk8sclient.EXPECT().ListMasterNodes().Return(GetKubeNodes(hostIds), nil).Times(1)
-			mockbmclient.EXPECT().UpdateHostStatus(Joined, hostIds[1]).Times(1)
+			mockbmclient.EXPECT().GetEnabledHostsNamesIds().Return(inventoryNamesIds, nil).Times(1)
+			mockk8sclient.EXPECT().ListMasterNodes().Return(GetKubeNodes(map[string]string{}), nil).Times(1)
+			kubeNamesIds = map[string]string{"node0": "7916fa89-ea7a-443e-a862-b3e930309f65"}
+			//hostIds := []string{"7916fa89-ea7a-443e-a862-b3e930309f65"}
```
Comment (tsorya) [inline src/installer/installer_test.go:132]: same

File: src/installer/installer_test.go [hunk 5]
```diff
@@ -109,13 +126,17 @@ var _ = Describe("installer HostRoleMaster role", func() {
...
+			mockk8sclient.EXPECT().ListMasterNodes().Return(GetKubeNodes(kubeNamesIds), nil).Times(1)
+			mockbmclient.EXPECT().UpdateHostStatus(Joined, inventoryNamesIds["node0"]).Times(1)
+			kubeNamesIds = map[string]string{"node0": "7916fa89-ea7a-443e-a862-b3e930309f65",
+				"node1": "eb82821f-bf21-4614-9a3b-ecb07929f238"}
+			//hostIds = []string{"7916fa89-ea7a-443e-a862-b3e930309f65", "eb82821f-bf21-4614-9a3b-ecb07929f238"}
```
Comment (tsorya) [inline src/installer/installer_test.go:137]: same

File: src/installer/installer.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply src/installer/installer.go:294]: in controller it is different, here we don't need try loop

Comment (tsorya) [reply src/installer/installer.go:294]: why not? sure we will get it in one try?

**Pull request merged**